canvaswidget.cpp

```diff
@@ -130,12 +130,16 @@ void CanvasWidget::mousePressEvent(QMouseEvent *event) {
 
     selected = NULL;
     transformation = NONE;
+    if (!shapes.size ())
+      return;
```

Comment:
this condition looks to do not have any sense b/c there will not be any loop if size is 0